### DISCUSSION TITLE
Tests fails

### DISCUSSION THREAD

**Discussion by MathiasWeisheit** [spatie/laravel-medialibrary#3123] [Q&A]
Hi, media library works well in my project.
But my featur test fails.:
        Storage::fake('public');
        $category = $this->modelClass::factory()->create();
        $url = $this->url.$category->id.'/cover';
        $file = UploadedFile::fake()->image('cover.jpg');
        $response = $this->json('POST', $url, ['image' => $file], $this->apiAuth());
The storage of the image fails, because of permissions of the file system.
Can someone help me?

**Comment by parallels999** [comment]
[laravel-medialibrary-tests](https://github.com/spatie/laravel-medialibrary/tree/main/tests)

**Comment by MathiasWeisheit** [comment]
Hi, it is strange:
the file created by medialibary has the owner root, but the files created by the Filesystem Class has nobody as owner.

 $file = UploadedFile::fake()->image('cover.jpg');
        $fs = Storage::fake('public');
        $fs->put('cover.jpg', $file);
        Storage::disk('public')->assertExists('cover.jpg');  //works

        $file = UploadedFile::fake()->image('cover2.jpg');

        $category = $this->modelClass::factory()->create();
        $url = $this->url.$category->id.'/testStorage';
        $response = $this->json('POST', $url, ['image' => $file], $this->apiAuth());
        Storage::disk('public')->assertExists('cover2.jpg');  //works

        $url = $this->url.$category->id.'/cover';
        $response = $this->json('POST', $url, ['image' => $file], $this->apiAuth());
        Storage::disk('public')->assertExists('cover.jpg');  //workd

        $response->assertStatus(200)->assertSeeText($file->name); //error 500

ErrorException: fopen(/app/storage/framework/testing/disks/public/1/cover2.jpg): Failed to open stream: Permission denied in /app/vendor/league/flysystem/src/Adapter/Local.php:157
  Stack trace:
  #0 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleError()

![grafik](https://user-images.githubusercontent.com/4928100/206004519-892c6438-df00-4553-80ee-dc782f261d10.png)

**Reply by MathiasWeisheit** [reply]
now i start my test as root user and it works fine
docker-compose exec -u root webserver php artisan test --filter
